Description: Disable connecting to hal, it's not used any more (that code is commented out).
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=593787
Author: Martin Pitt <martin.pitt@ubuntu.com>

diff -Nur -x '*.orig' -x '*~' gdm-2.29.5/daemon/gdm-local-display-factory.c gdm-2.29.5.new/daemon/gdm-local-display-factory.c
--- gdm-2.29.5/daemon/gdm-local-display-factory.c	2010-01-12 17:36:37.000000000 +0100
+++ gdm-2.29.5.new/daemon/gdm-local-display-factory.c	2010-01-14 16:41:03.405487225 +0100
@@ -525,29 +525,6 @@
         return TRUE;
 }
 
-static gboolean
-connect_to_hal (GdmLocalDisplayFactory *factory)
-{
-        factory->priv->proxy = dbus_g_proxy_new_for_name (factory->priv->connection,
-                                                          HAL_DBUS_NAME,
-                                                          HAL_DBUS_MANAGER_PATH,
-                                                          HAL_DBUS_MANAGER_INTERFACE);
-        if (factory->priv->proxy == NULL) {
-                g_warning ("Couldn't create proxy for HAL Manager");
-                return FALSE;
-        }
-
-        return TRUE;
-}
-
-static void
-disconnect_from_hal (GdmLocalDisplayFactory *factory)
-{
-        if (factory->priv->proxy == NULL) {
-                g_object_unref (factory->priv->proxy);
-        }
-}
-
 static GObject *
 gdm_local_display_factory_constructor (GType                  type,
                                        guint                  n_construct_properties,
@@ -565,8 +542,6 @@
                 g_warning ("Unable to register local display factory with system bus");
         }
 
-        connect_to_hal (factory);
-
         return G_OBJECT (factory);
 }
 
@@ -611,8 +586,6 @@
 
         g_hash_table_destroy (factory->priv->displays);
 
-        disconnect_from_hal (factory);
-
         G_OBJECT_CLASS (gdm_local_display_factory_parent_class)->finalize (object);
 }
 
